swift/IceStorm/clock/Sources/Publisher/main.swift

```diff
@@ -113,27 +98,34 @@ func run() -> Int32 {
         let dateFormatter = DateFormatter()
         dateFormatter.dateFormat = "dd/MM/YYYY HH:mm:ss"
 
-        let t = DispatchSource.makeTimerSource()
-        t.schedule(deadline: .now(), repeating: .seconds(1))
-        t.setEventHandler {
-            do {
-                try clock.tick(dateFormatter.string(from: Date()))
-            } catch is CommunicatorDestroyedException {
-                t.suspend()
-                exit(0)
-            } catch {
-                t.suspend()
-                print("Error: \(error)\n")
-                communicator.destroy()
-                exit(1)
+        // Send a tick every second until cancelled
+        let task = Task {
+            while true {
+                do {
+                    try await clock.tick(dateFormatter.string(from: Date()))
+                } catch let error as Ice.LocalException {
+                    print("tick invocation failed with: \(error.message)")
```

Comment:
The plain \(error) for a local exception displays the full C++ stack trace. We should double-check that's the desired behavior.

Reply:
I think it's undesirable.